### PR TITLE
Code clean up

### PR DESCRIPTION
**Pull request by ikarienator**
Warning free.

### CODE REVIEW

File: src/main/java/com/shapesecurity/functional/Thunk.java [hunk 1]
```diff
@@ -49,19 +49,17 @@ private Thunk(@NotNull Supplier<A> supplier) {
     @NotNull
     public final A get() {
         // Double locked.
-        if (this.value == null) {
+        A value = this.value;
```
Comment (ikarienator) [inline src/main/java/com/shapesecurity/functional/Thunk.java]: It is clear that `value` can only be turned from nullable to nonnull.

File: src/main/java/com/shapesecurity/functional/data/ImmutableList.java [hunk 2]
```diff
@@ -489,9 +492,9 @@ public final void foreach(@NotNull Effect<A> f) {
 
     /**
      * Tests using object identity whether this list contains the element <code>a</code>.
-     *
+     * <p>
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/functional/data/ImmutableList.java]: Looks like you're missing a closing tag to match this.

**Comment by michaelficarra** [comment]
LGTM otherwise.

**Comment by ikarienator** [comment]
I vaguely remember it is not allowed to have end p tags. It is not HTML
after all. Official [Oracle documents on javadocs](
http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format)
does not use end tags as well.

**Pull request merged**